www/install/php/Update-18.10.4_to_18.10.5.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/*
+ * Copyright 2005-2019 Centreon
+ * Centreon is developed by : Julien Mathis and Romain Le Merlus under
+ * GPL Licence 2.0.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation ; either version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ * PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, see <http://www.gnu.org/licenses>.
+ *
+ * Linking this program statically or dynamically with other modules is making a
+ * combined work based on this program. Thus, the terms and conditions of the GNU
+ * General Public License cover the whole combination.
+ *
+ * As a special exception, the copyright holders of this program give Centreon
+ * permission to link this program with independent modules to produce an executable,
+ * regardless of the license terms of these independent modules, and to copy and
+ * distribute the resulting executable under terms of Centreon choice, provided that
+ * Centreon also meet, for each linked independent module, the terms  and conditions
+ * of the license of that module. An independent module is a module which is not
+ * derived from this program. If you modify this program, you may extend this
+ * exception to your version of the program, but you are not obliged to do so. If you
+ * do not wish to do so, delete this exception statement from your version.
+ *
+ * For more information : contact@centreon.com
+ *
+ *
+ */
+
+/*
+ * Redo all PHP upgrades from 2.8.26 to 18.10.0 for people migrating from 2.8.27 before PR #7416
+ */
+$result = $pearDB->query(
+    "SELECT `value` FROM `informations` WHERE `key` IN ('appKey', 'isRemote', 'isCentral')"
+);
+
+if ($result->rowCount() == 0) {
+    $uniqueKey = md5(uniqid(rand(), true));
```

Comment:
MD5 is broken, uniqid() is not cryptographically secure and I do not even speak of rand().

WTF is this random number generation ? It is either over kill or totally inadequate. What is _appKey_ used for ?

Reply:
@ganoze I use exactly was it was plained for Remote Server in 18.10.0 (https://github.com/centreon/centreon/blob/master/www/install/php/Update-18.10.0.php)

Reply:
OK but why ? What are the security implications of the appKey ?

Reply:
Is mandatory to use Remote Server but not used really....

Reply:
Is this a joke ? I am reviewing some useless stuff ? If so, maybe instead of adding new code we might instead remove dead code ?

Reply:
The appKey is an extra identifier for remote server, because naigos_server table is related to remote_server by IP and sometimes that is not enough. This identifier is not used for authentication, validation, or other activity, and is nothing more than an additional ID of a server for internal recognition and furthermore is not sendable/readable by the end user, so for now I believe we can leave it as it is. I do agree the logic to generate keys can be changed in the future, but that is not the subject of this PR. The PR is blocking a release and should go through. Any changes to the key generations can be groomed as a change task and executed as soon as the priority comes to that.

Reply:
Still I am afraid that either this identifier is 1) useless or 2) overly complex or 3) used in some kind of unclear authorization scheme that is a potential security breach (most likely).

I am also afraid that if this PR go through, the technical debt added by it will never be processed.

Reply:
I will create a ticket to manage this technical debt